.github/workflows/tests.yml

```diff
@@ -151,52 +210,41 @@ jobs:
           draft: true
           delete-branch: true
 
-      # return failure if any checks failed
-      # return failure if any checks were cancelled
-      # loop again if any checks are still in progress
-      # loop again if no checks have completed yet
-      # return failure if checks are unchanged since last loop and the checks are empty
-      # return success if checks are unchanged since last loop and at least once check is completed
-      # (else) loop again if checks have changed since last loop (avoid race conditions)
+      # wait for checks register, complete and fail if any required checks failed
       - name: Wait for all checks to finish
         if: steps.cpr.outputs.pull-request-number != ''
         env:
           flowzone_re: "^Flowzone"
+          filter_re: "^policy-?bot"
+          GH_TOKEN: ${{ steps.gh_token.outputs.token || secrets.FLOWZONE_TOKEN }}
         run: |
-          while true
-          do
-            sleep 30
-
-            prev_checks="${checks}"
-
-            checks="$(gh pr status --json statusCheckRollup,number --jq '[.createdBy[] |
-              select(.number == ${{ steps.cpr.outputs.pull-request-number}}) | .statusCheckRollup[] |
-              select(.__typename == "CheckRun") |
-              select(.name | startswith("Flowzone")) |
-              {conclusion,status,name}]' || true)"
-
-            echo "${checks}" | jq
-
-            jq -r '.[].conclusion' <<< "${checks}" | grep -q '^FAILURE$' && exit 1
-            jq -r '.[].conclusion' <<< "${checks}" | grep -q '^CANCELLED$' && exit 1
-
-            jq -r '.[].status' <<< "${checks}" | grep -q '^IN_PROGRESS$' && continue
-            jq -r '.[].status' <<< "${checks}" | grep -q '^QUEUED$' && continue
-            jq -r '.[].status' <<< "${checks}" | grep -q '^PENDING$' && continue
-
-            if [ "${prev_checks}" == "${checks}" ]
-            then
-              [ $(jq 'length' <<< "${checks}") -gt 0 ] || exit 1
-              jq -r '.[].status' <<< "${checks}" | grep -q '^COMPLETED$' && exit 0
-            fi
+          while ! gh pr checks --required \
+            --repo ${{ matrix.repository }} \
+            ${{ steps.cpr.outputs.pull-request-number}} | grep -qE "${flowzone_re}"; do
```

Comment:
Is it okay if one Flowzone check has loaded, but not the others? I think visually I have seen some required checks take longer to show up than others, and I wonder if we wont end up in another race condition with GitHub.

Maybe I'm just paranoid from being burned too many times.

Reply:
@klutchell let's catch up on this - it's a wider issue I think than just this PR. You are right, checks registration lags.